.profile.d/heroku-metrics-daemon.sh

```diff
@@ -33,13 +33,21 @@ echo "agentmon setup took ${ELAPSEDTIME} seconds"
 
 AGENTMON_FLAGS=()
 
-if [[ -f build.sbt ]] || [[ -d target/resolution-cache ]]; then
-    unzip -qq bin/heroku-metrics-agent.jar 'javax/*' -d bin/ext/
-    export JAVA_OPTS="-javaagent:bin/heroku-metrics-agent.jar=cp=/app/bin/ext/ -Xbootclasspath/a:bin/heroku-metrics-agent.jar ${JAVA_OPTS}"
-    AGENTMON_FLAGS+=("-prom-url=http://localhost:${HEROKU_METRICS_PROM_PORT}${HEROKU_METRICS_PROM_ENDPOINT}")
-elif [[ -f pom.xml ]] || [[ -f build.gradle ]] || [[ -f project.clj ]] || [[ -d .jdk ]]; then
-    export JAVA_TOOL_OPTIONS="-javaagent:bin/heroku-metrics-agent.jar ${JAVA_TOOL_OPTIONS}"
-    AGENTMON_FLAGS+=("-prom-url=http://localhost:${HEROKU_METRICS_PROM_PORT}${HEROKU_METRICS_PROM_ENDPOINT}")
+
+if [[ -f bin/heroku-metrics-agent.jar ]]; then
+    if [[ -f build.sbt ]] ||
+       [[ -d target/resolution-cache ]]; then
+        unzip -qq bin/heroku-metrics-agent.jar 'javax/*' -d bin/ext/
+        export JAVA_OPTS="-javaagent:bin/heroku-metrics-agent.jar=cp=/app/bin/ext/ -Xbootclasspath/a:bin/heroku-metrics-agent.jar ${JAVA_OPTS}"
+        AGENTMON_FLAGS+=("-prom-url=http://localhost:${HEROKU_METRICS_PROM_PORT}${HEROKU_METRICS_PROM_ENDPOINT}")
+    elif [[ -f pom.xml ]] ||
+         [[ -f build.gradle ]] ||
+         [[ -f project.clj ]] ||
+         [[ -f target/dependency/webapp-runner.jar ]] ||
+         [[ -d .jdk ]]; then
```

Comment:
FWIW: This then includes jruby when on a heroku-16 stack.

Reply:
Yes, `.jdk` was always there. it should include JRuby. The addition is the line above it (webapp-runner).